lib/resque/data_store.rb

```diff
@@ -275,6 +276,10 @@ def all_heartbeats
         @redis.hgetall(HEARTBEAT_KEY)
       end
 
+      def acquire_pruning_dead_worker_lock(worker, expiry)
+        @redis.set(redis_key_for_worker_pruning, worker.to_s, :ex => expiry, :nx => true)
```

Comment:
I just realized that this command is only available in Redis version >= 2.6.12 and `redis-rb` version >= 3.05. Not sure what to do about that. It's possible to reimplement the same behaviour via a combination of `MULTI`, `SETNX`, `EXPIRE`, and `WATCH`, but that feels kinda hacky. Any suggestions? Could just disable this feature completely if the gem or the server version are too old. Thoughts?

Reply:
Could also do something slightly different:
```ruby
def acquire_pruning_dead_worker_lock(worker, expiry)
  setnx, expire = @redis.multi do
    @redis.setnx(redis_key_for_worker_pruning, worker.to_s)
    @redis.expire(redis_key_for_worker_pruning, expiry)
  end

  setnx
end
```
but the semantics are slightly different since this version bumps the expiry on every call while the other version only sets the expiry ones and then never bumps it until expires. The second version could lead to the key never expiring if workers reboot frequently (frequent deploys).

Reply:
Is it easy to get the version? I'm somewhat worried about 1000s of workers asking the server for its version on deploy.

Reply:
In the README it says:

```
Resque now supports Ruby 2.0.0 and above. Any future updates will not be guaranteed to work without defects on any Rubies older than 2.0.0. We will also only be supporting Redis 3.0 and above going forward.
```

Reply:
Yea good point. The Redis `INFO` command returns the server version. But I'm not sure how expensive that command is (probably much less expensive than doing the entire pruning though, but the concern remains).

Reply:
Support for old Redis was removed in this PR https://github.com/resque/resque/pull/1594

Reply:
Since that PR we only support `>= 2.6` [where `TIME` was introduced](https://redis.io/commands/time). 

I'd say you're good here, @fw42